sdk/core/azure-core/src/main/java/com/azure/core/credential/TokenRequestContext.java

```diff
@@ -50,4 +58,46 @@ public TokenRequestContext addScopes(String... scopes) {
         this.scopes.addAll(Arrays.asList(scopes));
         return this;
     }
+
+    /**
+     * Get the configured Http Client.
+     * @return the Http Client.
+     */
+    public HttpClient getHttpClient() {
+        return httpClient;
+    }
+
+    /**
+     * Get the configured log options.
+     * @return the log options.
+     */
+    public HttpLogOptions getLogOptions() {
```

Comment:
There are other policies that we are not including like retry policy, timeout policy etc. Any reason why only HttpLogOptions is included here? Can this be deferred, so we can think of all other policies that might be needed and add it when required?